pydantic/type_adapter.py

```diff
@@ -245,7 +245,10 @@ def __init__(self, type: Any, *, config: ConfigDict | None = None, _parent_depth
         try:
             validator = _getattr_no_parents(type, '__pydantic_validator__')
         except AttributeError:
-            validator = create_schema_validator(core_schema, core_config, config_wrapper.plugin_settings)
+            f = sys._getframe(1)
+            validator = create_schema_validator(
+                core_schema, f'{f.f_globals["__name__"]}:{type}', core_config, config_wrapper.plugin_settings
+            )
```

Comment:
instead of this, I think it would be better to add a parameter

```py
module: str = __name__,
```
argument if that works.

Reply:
If we have `TypeAdapter(List[str])` in `test.test_plugins`:

- Current `type_path` = `tests.test_plugins:typing.List[str]`
- By your suggested change, `type_path` = `pydantic.type_adapter:typing.List[str]`

Should we do the change?

Reply:
no shit, no I see the problem, using `__name__` doesn't work - either here or in `create_model` - you will need to use `get_frame()`.